src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -140,7 +144,30 @@ public function decode(string $data, string $format, array $context = []): mixed
             }
         }
 
-        // todo: throw an exception if the root node name is not correctly configured (bc)
+        // Validate root node existence and, if configured by the user, its name
+        $shouldValidateRootExists = (bool) ($context[self::VALIDATE_ROOT_NODE_EXISTS] ?? $this->defaultContext[self::VALIDATE_ROOT_NODE_EXISTS] ?? false);
+        if (!$rootNode instanceof \DOMNode) {
+            if ($shouldValidateRootExists) {
+                throw new NotEncodableValueException('Invalid XML data: no root node found.');
+            }
+
+            trigger_deprecation('symfony/serializer', '7.4', 'Decoding XML without a root node is deprecated and will throw a NotEncodableValueException in 8.0.');
+
+            // Return empty result for backward compatibility
```

Comment:
This comment doesn't bring added value. However, you may add the exception to throw with a comment so it's easy in 8.0 to deal with as we would only have to uncomment the exception throw